microsoft/MixedRealityToolkit-Unity
# Consistency in assembly and asmdef names

The change updates all assembly names and asmdef file names to match the following format:

Microsoft.MixedReality.Toolkit.<name[.subname]>

Due to file name and path lengths, the examples\demos assemblies were renamed to drop "Examples" from their names.

Testing performed
- Run solver demo scene in the editor, ensure
    - Object selection
    - Teleport
    - Boundary
    - Diagnostics
- Build with .net backend and compile in Visual Studio 2017
- Editor tests pass
